libcnb/src/layer/handling.rs

```diff
@@ -336,6 +345,20 @@ pub(crate) fn overwrite_layer_exec_d_programs<P: AsRef<Path>>(
     Ok(())
 }
 
+fn write_layer_metadata<M: Serialize, P: AsRef<Path>>(
+    layers_dir: P,
```

Comment:
Can you give a reasoning why you would prefer `impl` here?

Reply:
Mentally the two are synonymous, but `impl TraitName` is easier for me to read. 

Here's roughly how I translate the experiences of reading those two different styles. 

- `layers_dir: impl AsRef<Path>: "I saw a dog that is blue" has all of the information I need in the order I need 
- `<P: AsRef<path>>`  ... `layers_dir: P` "I saw a dog B" ... look for the definition of B going backwards "B is blue" then put them back together "I saw a dog B, which means it is blue."  Or alternatively "B will mean blue" okay, I'll hold onto this and don't know when it will be useful..."I saw a dog B" oh right, I remember that I know what B is already "therefore it is blue." 

It's the difference between reading `3 5 +` (reverse polish notation) and `3 + 5`. Both have the same information, but one requires more temporary memory to grok.

I was curious if there are practical implications towards using this versus a generic so I looked up the differences and found this https://stackoverflow.com/a/47515540. The downside is that you cannot explicitly specify the type in the function signature, which I think is perfectly fine here.

Reply:
I find the `impl` version easier to read too fwiw.

I kinda see the `P: AsRef<Path>` notation as an advanced notation needed only in certain situations (where `P` is needed in multiple places etc), and so in general when writing things like this I would lean towards using the `impl` form unless it otherwise wasn't viable.

Reply:
Since we're talking preferences here and this PR is not changing how we do these things in this existing file - can we move this discussion/change request of of this PR?